Validate Polaris Web Components for App Home Surface

billfienberg opened the pull request.

nelsonwittwer commented on src/validations/polarisWebComponents.ts:
```diff
@@ -0,0 +1,188 @@
+  // Validate against @shopify/app-bridge-ui-types directly
+
+  // Check for Shopify UI components (s-* pattern)
+  const shopifyComponentPattern = /<s-[a-z-]+/gi;
```
I'd imagine using the Typescript library instead of pattern matching here would give us better results in the short and long run. Have you explored that at all yet?

billfienberg commented:
## 🎩 Tophat Take 2

### Prompt

```
Replace Polaris React Components with Polaris Web Components
```

### Model: claude-4-sonnet

#### Result

✅* With a single prompt, on the first try, the MCP did all of the following:

1. Added the required script tag to `app/root.tsx`
1. Installed TypeScript types (`@shopify/app-bridge-ui-types`)
1. Updated tsconfig.json to include the new types
1. Replaced Polaris React components with Polaris Web Components
1. Replaced App Bridge React components with App Bridge Web Components
1. Called `validate_polaris_wc_codeblocks` MCP tool

*`validate_polaris_wc_codeblocks` tool wasn't called until the very end. 

### Visual Aids

#### Descript

#### LLM-generated TODOs

<img width="727" alt="image" src="https://github.com/user-attachments/assets/43169445-7723-4bae-a827-322b9c9aa1e5" />

#### `get_started` and `fetch_docs_by_path`

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/6ed16714-867f-4dcc-99eb-4eb8e1bc52d4" />

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/ece180c8-c6c0-4090-8851-f2bac281f8ce" />

#### `validate_polaris_wc_codeblocks` Params

```json
{
  "codeblocks": [
    "```html\n<s-page>\n  <ui-title-bar title=\"Remix app template\">\n    <s-button variant=\"primary\" onClick={generateProduct} slot=\"actions\">\n      Generate a product\n    </s-button>\n  </ui-title-bar>\n  <s-stack gap=\"large\">\n    <s-grid gridTemplateColumns=\"2fr 1fr\" gap=\"large\">\n      <s-section>\n        <s-stack gap=\"large\">\n          <s-stack gap=\"base\">\n            <s-heading>Congrats on creating a new Shopify app 🎉</s-heading>\n            <s-text>\n              This embedded app template uses\n              <s-link href=\"https://shopify.dev/docs/apps/tools/app-bridge\" target=\"_blank\">\n                App Bridge\n              </s-link>\n              interface examples like an <s-link href=\"/app/additional\">additional page in the app nav</s-link>, as\n              well as an\n              <s-link href=\"https://shopify.dev/docs/api/admin-graphql\" target=\"_blank\">\n                Admin GraphQL\n              </s-link>\n              mutation demo, to provide a starting point for app development.\n            </s-text>\n          </s-stack>\n        </s-stack>\n      </s-section>\n    </s-grid>\n  </s-stack>\n</s-page>\n```"
  ]
}
```

<img width="1624" alt="image" src="https://github.com/user-attachments/assets/bf6df308-ece6-4c20-a20a-8ba0a613e7dd" />

#### Summary

<img width="725" alt="image" src="https://github.com/user-attachments/assets/b642fcaa-6e2d-4084-8523-b2ce333301eb" />

<details>
<summary>Transcript</summary>

I have successfully replaced all Polaris React Components with Polaris Web Components in your Shopify app. Here's what was accomplished:
✅ Changes Made
Set up Polaris Web Components infrastructure:
Added the required script tag to app/root.tsx
Installed TypeScript types (@shopify/app-bridge-ui-types)
Updated tsconfig.json to include the new types
Replaced all React components with web components:
Page → s-page
Layout → s-grid with proper grid columns
Text (headings) → s-heading
Text (regular) → s-text
Card → s-section (provides card appearance)
Button → s-button
BlockStack → s-stack
InlineStack → s-stack with direction="inline"
Box → s-box
List → s-unordered-list
Link → s-link
TitleBar → ui-title-bar
Fixed property mappings:
url → href for links
variant="plain" → variant="secondary" for buttons
align="space-between" → justifyContent="space-between"
display="inline" → direction="inline"
Fixed camelCase properties like gridTemplateColumns and borderRadius
✅ Validation Results
Build Status: ✅ Successful compilation with no errors
Component Validation: ✅ All Polaris Web Components validated successfully
TypeScript: ✅ All type errors resolved
</details>

nelsonwittwer commented:
@billfienberg - Will you rebase this PR from [this other branch](https://github.com/Shopify/dev-mcp/pull/57)? It looks like you are a few stacked PRs behind in this PR 🙏
